bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java

```diff
@@ -68,189 +65,180 @@
+            LocalDate today = LocalDate.now();
+            return getConsumptionData(LocalDate.of(today.getYear() - 1, 1, 1), today);
+        });
     }
 
     @Override
     public void initialize() {
         logger.debug("Initializing Linky handler.");
         updateStatus(ThingStatus.UNKNOWN);
-        scheduler.submit(this::login);
-
-        final LocalDateTime now = LocalDateTime.now();
-        final LocalDateTime nextDayFirstTimeUpdate = now.plusDays(1).withHour(REFRESH_FIRST_HOUR_OF_DAY)
-                .truncatedTo(ChronoUnit.HOURS);
-        refreshJob = scheduler.scheduleWithFixedDelay(this::updateData,
-                ChronoUnit.MINUTES.between(now, nextDayFirstTimeUpdate) % REFRESH_INTERVAL_IN_MIN + 1,
-                REFRESH_INTERVAL_IN_MIN, TimeUnit.MINUTES);
-    }
 
-    private static Builder getLoginBodyBuilder() {
-        return new FormBody.Builder().add("encoded", "true").add("gx_charset", "UTF-8").add("SunQueryParamsString",
-                Base64.getEncoder().encodeToString("realm=particuliers".getBytes(StandardCharsets.UTF_8)));
-    }
+        LinkyConfiguration config = getConfigAs(LinkyConfiguration.class);
+        scheduler.submit(() -> {
+            try {
+                enedisApi = new EnedisHttpApi(config, gson, httpClient);
 
-    private synchronized boolean login() {
-        logger.debug("login");
+                updateStatus(ThingStatus.ONLINE);
 
-        LinkyConfiguration config = getConfigAs(LinkyConfiguration.class);
-        Request requestLogin = new Request.Builder().url(LOGIN_BASE_URI)
-                .post(getLoginBodyBuilder().add("IDToken1", config.username).add("IDToken2", config.password).build())
-                .build();
-        try (Response response = client.newCall(requestLogin).execute()) {
-            if (response.isRedirect()) {
-                logger.debug("Response status {} {} redirects to {}", response.code(), response.message(),
-                        response.header("Location"));
-            } else {
-                logger.debug("Response status {} {}", response.code(), response.message());
+                if (thing.getProperties().isEmpty()) {
+                    Map<String, String> properties = discoverAttributes();
+                    updateProperties(properties);
+                }
+
+                prmId = thing.getProperties().get(PRM_ID);
+                userId = thing.getProperties().get(USER_ID);
+
+                final LocalDateTime now = LocalDateTime.now();
+                final LocalDateTime nextDayFirstTimeUpdate = now.plusDays(1).withHour(REFRESH_FIRST_HOUR_OF_DAY)
+                        .truncatedTo(ChronoUnit.HOURS);
+
+                refreshJob = scheduler.scheduleWithFixedDelay(this::updateData,
+                        ChronoUnit.MINUTES.between(now, nextDayFirstTimeUpdate) % REFRESH_INTERVAL_IN_MIN + 1,
+                        REFRESH_INTERVAL_IN_MIN, TimeUnit.MINUTES);
```

Comment:
If `dispose()` is called immediately after `initialize()` it might be possible that this field is populated after `dispose()` finishes running.

The `refreshJob` field should be populated as part of `initialize()`, not asynchronously. The same could be said of any other field that requires cleanup during `dispose` (e.g. enedisApi).

Reply:
Ok, clear and obvious, I'll apply, but IIRW, I've been asked on some other code review to use scheduler.submit to avoid having initialize() handling a long process.

Reply:
> Ok, clear and obvious, I'll apply, but IIRW, I've been asked on some other code review to use scheduler.submit to avoid having initialize() handling a long process.

That would be the correct thing to do, yes. Https requests should not be done as part of initialization, but as part of an asynchronous task kicked off by initialization. Your original implementation was correct in that regard. My only real concern from your original implementation was that `refreshJob` was initialized asynchronously, but disposed synchronously. Your new implementation executes everything during `initialize()` which was not my intention to recommend and would very much like you to make it asynchronous again. I just want the `refreshJob` task to be created as part of initialize, but keep everything else asynchronous as it was before.